### PR TITLE
feat: radio uses props of wrapping form control

### PR DESCRIPTION
**Pull request by jmiazga**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes
      /start features)

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Fixes: #2631

## What is the new behavior?

`FormControl` propagates props down to `Radio` children.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by jvmazagao — APPROVED** [review]

File: packages/radio/src/use-radio.ts [hunk 1]
```diff
@@ -132,46 +132,93 @@ export function useRadio(props: UseRadioProps = {}) {
...
+        "required",
+        "aria-invalid",
+        "aria-required",
+        "aria-readonly",
+        "aria-describedby",
+        "onFocus",
+        "onBlur",
+      ])
+
+      const trulyDisabled = ownProps.disabled && !isFocusable
```
Comment (with-heart) [inline packages/radio/src/use-radio.ts:184]: `isSeriouslyForRealDisabled` 😆

**Review by with-heart — APPROVED** [review]

File: packages/radio/src/use-radio.ts [hunk 2]
```diff
@@ -61,7 +63,7 @@ export interface UseRadioProps {
    */
   isInvalid?: boolean
   /**
-   * If `true`, the radio button will be invalid. This sets `aria-invalid` to `true`.
+   * If `true`, the radio button will be required. This sets `aria-invalid` to `true`.
```
Comment (mustafawm) [inline packages/radio/src/use-radio.ts:66]: I think this still needs updating, compared to [<Input />'s isRequired prop](https://chakra-ui.com/docs/form/input#props)
"If `true`, the form control will required. This has 2 side effects: - The `FormLabel` will show a required indicator - The form element (e.g, Input) will have `aria-requred` set to `true`"

Comment (jmiazga) [reply packages/radio/src/use-radio.ts:66]: Yeah you are right, want to make a quick PR for this?

Comment (mustafawm) [reply packages/radio/src/use-radio.ts:66]: Done #2674

**Pull request merged**